src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,34 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface Props {
+  selectedTodo: Todo;
+  onResetTodo: () => void
+}
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, onResetTodo }) => {
+  const [loading, setLoading] = useState(true);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+
+  const { id, title, completed } = selectedTodo;
+
+  useEffect(() => {
+    getUser(selectedTodo.userId)
+      .then(setSelectedUser)
+      .finally(() => setLoading(false));
+  }, [selectedTodo?.userId]);
 
-export const TodoModal: React.FC = () => {
   return (
-    <div className="modal is-active" data-cy="modal">
+    <div
+      className="modal is-active"
+      data-cy="modal"
+    >
       <div className="modal-background" />
 
-      {true ? (
+      {loading ? (
```

Comment:
```suggestion
      {!selectedUser? (
```
you can just use `selectedUser` and you won't need `loading`